core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java

```diff
@@ -61,7 +62,12 @@ public String metadataLocation() {
   }
 
   public TableMetadata tableMetadata() {
-    return TableMetadata.buildFrom(metadata).withMetadataLocation(metadataLocation).build();
+    if (null == metadataWithLocation) {
```

Comment:
we were rebuilding the metadata on every call to `tableMetadata()` which was required when java reflection was used for JSON deserialiazion. However, now this isn't needed anymore, so we can compute the metadata with the location only once